build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/transport/TransportVersionResourcesService.java

```diff
@@ -218,10 +224,14 @@ List<TransportVersionUpperBound> getUpperBoundsFromUpstream() throws IOException
     }
 
     /** Write the given upper bound to a file in the transport resources */
-    void writeUpperBound(TransportVersionUpperBound upperBound) throws IOException {
+    void writeUpperBound(TransportVersionUpperBound upperBound, boolean stageInGit) throws IOException {
         Path path = transportResourcesDir.resolve(getUpperBoundRelativePath(upperBound.name()));
         logger.debug("Writing upper bound [" + upperBound + "] to [" + path + "]");
         Files.writeString(path, upperBound.definitionName() + "," + upperBound.definitionId().complete() + "\n", StandardCharsets.UTF_8);
+
+        if (stageInGit) {
+            gitCommand("add", path.toString());
```

Comment:
Is this really necessary? I mean we don't do this for any other kind of generation tasks. Can we not just rely on folks to do this themselves just as with any other local changes that need to be committed/pushed?

Reply:
> I mean we don't do this for any other kind of generation tasks

We _could_ consider doing this for other files...

Reply:
I worry if we don't add the files it would actually be more confusing since they are absolutely necessary to run ES so I'm +1 on this.

Reply:
I don't have super strong feelings on this but I'm generally -1 on things that muck with folks' local git state. There's just so many unique workflows people use and we just don't know when we might be stepping on those.

Reply:
Sure maybe, but we don't do this when you run `spotlessApply` or other tasks that mutate local version controlled files. It's expected that after that you do all the normal `git` stuff yourself.

To me it just seems kind of unnecessary, deviates from what we do for other tasks, and there's not implicit expectation that the build would be doing this for you so in the worst case it would be surprising to folks.

> We could consider doing this for other files...

By this logic why isn't this the default behavior for _any_ call to this task? Again, it's just no consistent with any other similar behavior we have, in which case I'm apt to just have it work like everything else.

Reply:
I suppose the question is "how magic is the tool/transport versions". We don't auto-stage `spotlessApply` because it's touching code we wrote. No magic. But we *do* auto-stage it when CI does `spotlessApply` for us.

If transport versions are managed by the tool almost entirely, then, yeah, staging is nice and useful.

As far as `commit -a` - whenever I get a merge conflict I manually `git add` each resolved file one by one on the command line. But I'm old now. I have no idea what most folks do.

I'd be really creeped out if it auto-staged files it isn't managing. But that's not this.

Reply:
> If transport versions are managed by the tool almost entirely, then, yeah, staging is nice and useful.

That's a fair distinction. In which case to my previous comment, I would then expect it to auto stage for _any_ changes it makes to files it manages, not just when passing `--resolve-conflict`.

Reply:
I was just thinking maybe it should auto-stage for anything involving the internal transport version files because we really don't expect or want users to have to consider these at all.

Reply:
I don't think we expect folks to understand all those distinctions. Basically we don't expect folks to every manually touch these files. If that's the case, then I agree with Nik that the task should just "own" these files and handle staging them in all scenarios if that's what we're gonna do.